python/cudf/cudf/core/column/lists.py

```diff
@@ -360,14 +364,35 @@ def get(self, index: int) -> ParentType:
         1    5
         2    6
         dtype: int64
+
+        >>> s = cudf.Series([[1, 2], [3, 4, 5], [4, 5, 6]])
+        >>> s.list.get(2)
+        0    <NA>
+        1       5
+        2       6
+        dtype: int64
+
+        >>> s = cudf.Series([[1, 2], [3, 4, 5], [4, 5, 6]])
+        >>> s.list.get(2, default=0)
+        0   0
+        1   5
+        2   6
+        dtype: int64
         """
-        min_col_list_len = self.len().min()
-        if -min_col_list_len <= index < min_col_list_len:
-            return self._return_or_inplace(
-                extract_element(self._column, index)
-            )
-        else:
-            raise IndexError("list index out of range")
+        out = extract_element(self._column, index)
+
+        if not (default is None or default is cudf.NA):
```

Comment:
```suggestion
        if default not in {None, cudf.NA}:
```

Reply:
Reverted this for now. This doesn't quite work because of reasons I fully haven't understood. Some observations:

```
[1, 2]  in {None,}  # raises
[1, 2] in (None,)  # returns False
[1, 2] in {None, cudf.NA}  # raises
[1, 2] in (None, cudf.NA)  # returns False
0 in {None, cudf.NA}  # raises
0 in (None, cudf.NA)  # returns False
```

Reply:
Members of a set must be hashable. A list is not hashable. That’s probably the reason. 🥲 You can use a tuple though:
```suggestion
        if default in (None, cudf.NA):
```

I suggested the set because it’s a tiny bit faster than the tuple for containment checks. Unfortunately I didn’t consider hashability.

Reply:
That still doesn't work for the case e.g., `0 in [cudf.NA,]` -- note that the issue seems to be in Pandas:

```python
In [38]: 0 in [pd.NA,]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [38], in <cell line: 1>()
----> 1 0 in [pd.NA,]

File ~/compose/etc/conda/cuda_11.5/envs/rapids/lib/python3.8/site-packages/pandas/_libs/missing.pyx:446, in pandas._libs.missing.NAType.__bool__()

TypeError: boolean value of NA is ambiguous
```